gnmi_subscribe/gnmi_subscribe.go

```diff
@@ -176,11 +241,55 @@ func assembleSubscriptions(paths []*pb.Path) ([]*pb.Subscription, error) {
 		subscription := &pb.Subscription{
 			Path:              path,
 			Mode:              subscriptionMode,
-			SampleInterval:    *sampleInterval,
+			SampleInterval:    sampleInterval,
 			SuppressRedundant: *suppressRedundant,
 			HeartbeatInterval: *heartbeatInterval,
 		}
 		subscriptions = append(subscriptions, subscription)
 	}
 	return subscriptions, nil
 }
+
+func subscriptionMode(subscriptionPoll, subscriptionOnce bool) (gnmi.SubscriptionList_Mode, error) {
+	switch {
+	case subscriptionPoll && subscriptionOnce:
+		return 0, errors.New("Only one of -once and -poll can be set")
```

Comment:
Is there a reference type you can return here rather than value? pb.SubscriptionList_UNKNOWN could be created if it doesn't exist?